[Docs] Add story: shopping cart

pvcarrera opened the pull request:
This PR copies shopping cart exercise over from the research repository. I think this exercise is complex enough to teach a bunch of things 😄 But probably the main is dealing with the state of the cart

pvcarrera commented:
I run `./bin/format.sh` but prettier is still complaining, I'm not sure what I'm missing here 😄

kotp commented on reference/stories/shopping-cart.md:
```diff
@@ -0,0 +1,37 @@
+# Shopping cart exercise
+
+There are multiple ways of solving this exercise. It's probably necessary to master some basic concepts like: strings, hashes or collections before this exercise can be tackle.
```
```suggestion
There are multiple ways of solving this exercise. It's probably necessary to master some basic concepts like: strings, hashes or collections before this exercise can be tackled.
```

pvcarrera replied:
Thanks @kotp !

wolf99 commented:
Looks good!

SleeplessByte commented on reference/stories/shopping-cart.md:
```diff
@@ -0,0 +1,37 @@
+- `total_amount`: The value of the items in the cart as a number with two decimal places (e.g. 3.99).
+- `items_list`: A list of the names of the items in the cart, ordered alphabetically, with no duplicates, separated by a comma and a space.
+
+For example:
```
I'd leave the ruby code out, but instead link to the ruby example implementation / exercise, like [this](https://github.com/exercism/v3/blob/master/reference/stories/arrays.magician-card-stack.md#tasks)

ErikSchierboom commented:
@pvcarrera Thanks for doing this! I've made some consistency changes to the story format this morning, which can be found [in this PR](https://github.com/exercism/v3/pull/1408). Note that the only section that is required is the `## Story` section. The other sections are all optional.

pvcarrera commented:
@ErikSchierboom Thanks for keeping the repos consistent 🙇‍♂️ I adjusted the format and added the exercise extension. I'm still not sure which concept/concepts will go together with this exercise, for now, I added `state` as the concept on the stories menu.
